core/src/test/java/org/apache/accumulo/core/client/impl/ThriftTransportKeyTest.java

```diff
@@ -74,28 +84,14 @@ public void testSslAndSaslErrors() {
   public void testConnectionCaching() throws IOException, InterruptedException {
     UserGroupInformation user1 = UserGroupInformation.createUserForTesting("user1", new String[0]);
     final KerberosToken token = EasyMock.createMock(KerberosToken.class);
-    final ClientConfiguration clientConf = ClientConfiguration.loadDefault();
-    // The primary is the first component of the principal
-    final String primary = "accumulo";
-    clientConf.withSasl(true, primary);
 
     // A first instance of the SASL cnxn params
     SaslConnectionParams saslParams1 = user1
-        .doAs(new PrivilegedExceptionAction<SaslConnectionParams>() {
-          @Override
-          public SaslConnectionParams run() throws Exception {
-            return new SaslConnectionParams(clientConf, token);
-          }
-        });
+        .doAs((PrivilegedExceptionAction<SaslConnectionParams>) () -> createSaslParams(token));
 
     // A second instance of what should be the same SaslConnectionParams
     SaslConnectionParams saslParams2 = user1
-        .doAs(new PrivilegedExceptionAction<SaslConnectionParams>() {
-          @Override
-          public SaslConnectionParams run() throws Exception {
-            return new SaslConnectionParams(clientConf, token);
-          }
-        });
+        .doAs((PrivilegedExceptionAction<SaslConnectionParams>) () -> createSaslParams(token));
```

Comment:
Is the explicit type required in these lambdas? The lambda is better than the anonymous class, but I'm wondering if it can be simplified even further.

Reply:
It has be explicit or the code won't build.